plugins/org.locationtech.udig.render.gridcoverage.basic.test/src/org/locationtech/udig/render/internal/gridcoverage/basic/BasicGridCoverage2DTestReader.java

```diff
@@ -0,0 +1,104 @@
+package org.locationtech.udig.render.internal.gridcoverage.basic;
+
+import java.awt.Rectangle;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.coverage.grid.io.AbstractGridCoverage2DReader;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.parameter.Parameter;
+import org.geotools.parameter.ParameterGroup;
+import org.opengis.coverage.grid.Format;
+import org.opengis.parameter.GeneralParameterValue;
+import org.opengis.parameter.ParameterValueGroup;
+
+public class BasicGridCoverage2DTestReader extends AbstractGridCoverage2DReader {
+
+    private int reads;
+    
+    public int getReads() 
+    {
+        return reads;
+    }
+
+
+    public BasicGridCoverage2DTestReader() 
+    {
+        this.reads = 0;
+    }
+    
+    
+    @Override
+    public Format getFormat() {
+        return new Format() {
+
+            @Override
+            public String getDescription() {
+                return null;
+            }
+
+            @Override
+            public String getDocURL() {
+                return null;
+            }
+
+            @Override
+            public String getName() {
+                return null;
+            }
+
+            @Override
+            public ParameterValueGroup getReadParameters() {
+                Map<String, Object> paramMap = new HashMap<String, Object>();
+                paramMap.put("name", "BasicGridCoverage2DFormat");  //$NON-NLS-1$//$NON-NLS-2$
+                paramMap.put(AbstractGridFormat.READ_GRIDGEOMETRY2D.getName().toString(),new GridGeometry2D(new Rectangle(), new Rectangle()));
+                
+/*                GridEnvelope2D gridEnvelope = new GridEnvelope2D(0, 0, mapDisplay.getWidth(), mapDisplay.getHeight());
```

Comment:
Do we need the commented lines on intial commit? Are you kind to remove these?

Reply:
Sorry for the bad code formatting. Hope everything is fine now :)